crates/forge/src/runner.rs

```diff
@@ -158,6 +158,11 @@ impl<'a> ContractRunner<'a> {
             U256::ZERO,
             Some(&self.mcr.revert_decoder),
         );
+
+        if let Err(EvmError::Execution(err)) = &deploy_result {
+            return Ok(TestSetup::failed(format!("contract deployment failed: {err}")));
```

Comment:
maybe rename `contract deployment failed` to `test contract deployment failed` for clarity?